### PR TITLE
Add revision list view to server

### LINKED ISSUE #775: Server: Add endpoint giving revisions it can serve
Add an `GET` endpoint which lists out the versions of the project it can find on the disk. It should be as simple as listing the directory one up of `MODEL_COLLECTION_DIR`. 
Also consider changing it a bit so instead of taking in `MODEL_COLLECTION_DIR` which is `/gordo/models/{{project_name}}/{{project_version}}` the server takes in `/gordo/models/{{project_name}}` and then a second parameter which is `default_version` ? Dealers choice. 

Feel free to put a TTL cache with e.g `2s` timeout on it.

### PR DESCRIPTION
**Pull request by milesgranger**
Will close #775 

List project revisions the server can serve, including the latest

### CODE REVIEW

File: gordo_components/server/views/base.py [hunk 1]
```diff
@@ -252,6 +252,27 @@ def get(self, gordo_project: str):
...
+        try:
+            available_revisions = os.listdir(os.path.join(collection_dir, ".."))
+        except FileNotFoundError:
+            logger.error(
+                f"Attempted to list directories above {collection_dir} but failed with: {traceback.format_exc()}"
+            )
+            available_revisions = [collection_dir]
+
+        return jsonify(
+            {"latest": collection_dir, "available-revisions": available_revisions}
```
Comment (epa095) [inline gordo_components/server/views/base.py:272]: 👍

Comment (milesgranger) [reply gordo_components/server/views/base.py:272]: :hankey: :toilet: ?

Comment (epa095) [reply gordo_components/server/views/base.py:272]: :-1: :tv: :baby:

**Review by EinarJohnsen — APPROVED** [review]
LGTM :snowman_with_snow:

**Pull request merged**